### PR TITLE
Nisha | Test | Provide support of Entries export in Giddh

### PR DESCRIPTION
**Pull request by nisha-walkover**
https://app.clickup.com/t/86cvqx9c3

### CODE REVIEW

File: apps/web-giddh/src/app/services/apiurls/daybook.api.ts [hunk 1]
```diff
@@ -1,4 +1,5 @@
 export const DAYBOOK_SEARCH_API = {
     SEARCH: 'company/:companyUniqueName/daybook?page=:page&count=:count&from=:from&to=:to', // get call
     EXPORT: 'company/:companyUniqueName/export-daybook-v2?page=:page&count=:count&from=:from&to=:to&format=:format&type=:type&sort=:sort', // get call
+    ENTRIES_EXPORT: 'company/:companyUniqueName/exports?output=:output', // get call
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/services/apiurls/daybook.api.ts]: remove comma

File: apps/web-giddh/src/app/daybook/export-daybook/export-daybook.component.ts [hunk 2]
```diff
@@ -19,6 +19,7 @@ export class ExportDaybookComponent implements OnInit, OnDestroy {
     public emailTypeSelected: string = '';
     public emailTypeMini: string = '';
     public emailTypeDetail: string;
+    public emailTypeExpanded: string;
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/daybook/export-daybook/export-daybook.component.ts:25]: add doucmentation

File: apps/web-giddh/src/app/daybook/daybook.component.ts [hunk 3]
```diff
@@ -379,55 +383,77 @@ export class DaybookComponent implements OnInit, OnDestroy {
...
+                                    this.toasterService.showSnackBar("success", response?.body?.file);
+                                } else {
+                                    let blob = this.generalService.base64ToBlob(response?.body?.file, response?.queryString?.requestType, 512);
+                                    let type = response?.queryString?.requestType === 'application/pdf' ? '.pdf' : '.xls';
+                                    saveAs(blob, 'Daybook' + type);
+                                }
+                            } else {
+                                this.toasterService.showSnackBar("error", response?.message);
+                            }
+                        });
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/daybook/daybook.component.ts:423]: check this code with main stage branch

File: apps/web-giddh/src/app/daybook/daybook.component.ts [hunk 4]
```diff
@@ -146,6 +146,10 @@ export class DaybookComponent implements OnInit, OnDestroy {
         let hasParticularSelected = this.lc.blankLedger.transactions?.filter(txn => txn?.particular);
         return (hasParticularSelected?.length) ? true : false;
     }
+    /** This will use for export as file type */
+    public fileTypeExpand: string = 'CSV';
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/daybook/daybook.component.ts]: check this code

**Review by dilpreetsinghofficial — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/daybook/export-daybook/export-daybook.component.ts (hunk 2 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/daybook/export-daybook/export-daybook.component.ts:25]: done

File: apps/web-giddh/src/app/daybook/daybook.component.ts (hunk 4 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/daybook/daybook.component.ts]: this variable was not in use so I removed variable

File: apps/web-giddh/src/app/services/apiurls/daybook.api.ts (hunk 1 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/apiurls/daybook.api.ts]: done

File: apps/web-giddh/src/app/daybook/daybook.component.ts (hunk 3 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/daybook/daybook.component.ts:423]: done

File: apps/web-giddh/src/app/services/daybook.service.ts [hunk 5]
```diff
@@ -86,4 +86,29 @@ export class DaybookService {
         }),
             catchError((e) => this.errorHandler.HandleCatch<DayBookResponseModel, DayBookRequestModel>(e, request)));
     }
+
+    /**
+     * For Export Daybook expanded
+     * @param request 
+     * @memberof DaybookService 
+     */
+    public ExportDaybookExpandedPost(request: any, branchUniqueName: string): Observable<BaseResponse<any, any>> {
```
Comment (dilpreetsinghofficial) [inline apps/web-giddh/src/app/services/daybook.service.ts]: make camel case format

File: apps/web-giddh/src/app/daybook/daybook.component.ts [hunk 6]
```diff
@@ -376,58 +378,80 @@ export class DaybookComponent implements OnInit, OnDestroy {
     public hideExportDaybookModal(response: any) {
         this.modalDialogRef.close();
         if (response !== 'close') {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/daybook/daybook.component.ts:380]:
```suggestion
        if (response && response !== 'close') {
```

File: apps/web-giddh/src/app/daybook/export-daybook/export-daybook.component.ts (hunk 2 above)
Comment (dvCodeWorld) [reply apps/web-giddh/src/app/daybook/export-daybook/export-daybook.component.ts:25]: Write documentation for  each line of variable declaration

File: apps/web-giddh/src/app/services/daybook.service.ts [hunk 7]
```diff
@@ -86,4 +86,29 @@ export class DaybookService {
         }),
             catchError((e) => this.errorHandler.HandleCatch<DayBookResponseModel, DayBookRequestModel>(e, request)));
     }
+
+    /**
+     * For Export Daybook expanded
+     * @param request 
+     * @memberof DaybookService 
+     */
+    public exportDaybookExpandedPost(request: any, branchUniqueName: string): Observable<BaseResponse<any, any>> {
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/daybook.service.ts:96]: branchUniqueName: string

this parameter is not in you function doc, add this too

File: apps/web-giddh/src/app/services/daybook.service.ts [hunk 8]
```diff
@@ -86,4 +86,29 @@ export class DaybookService {
...
+        this.companyUniqueName = this.generalService.companyUniqueName;
+        let url = this.config.apiUrl + DAYBOOK_SEARCH_API.ENTRIES_EXPORT
+            ?.replace(':companyUniqueName', encodeURIComponent(this.companyUniqueName))
+            ?.replace(':output', request.fileType?.toString())
```
Comment (dvCodeWorld) [inline apps/web-giddh/src/app/services/daybook.service.ts]:
```suggestion
            ?.replace(':output', request.fileType?.toString());
```

**Review by dvCodeWorld — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/daybook/daybook.component.ts (hunk 6 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/daybook/daybook.component.ts:380]: This code is not written by me it was written by someone five years ago.

File: apps/web-giddh/src/app/daybook/export-daybook/export-daybook.component.ts (hunk 2 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/daybook/export-daybook/export-daybook.component.ts:25]: done

File: apps/web-giddh/src/app/services/daybook.service.ts (hunk 7 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/daybook.service.ts:96]: done

File: apps/web-giddh/src/app/services/daybook.service.ts (hunk 8 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/services/daybook.service.ts]: done

**Review by dvCodeWorld — APPROVED** [review]

**Review by dilpreetsinghofficial — APPROVED** [review]

File: apps/web-giddh/src/app/daybook/daybook.component.ts [hunk 9]
```diff
@@ -376,58 +378,80 @@ export class DaybookComponent implements OnInit, OnDestroy {
...
+                exportBodyRequestObj.from = this.daybookQueryRequest.from;
+                exportBodyRequestObj.to = this.daybookQueryRequest.to;
+                exportBodyRequestObj.fileType = this.fileTypeExtension;
+                exportBodyRequestObj.exportType = "ENTRIES_EXPORT";
+                let branchUniqueName = this.generalService.currentBranchUniqueName ? this.generalService.currentBranchUniqueName : this.currentBranch ? this.currentBranch?.uniqueName : "";
+                this.daybookService.exportDaybookExpandedPost(exportBodyRequestObj, branchUniqueName).pipe(takeUntil(this.destroyed$)).subscribe(response => {
                     if (response?.status === 'success') {
                         if (response?.body?.type === "message") {
-                            this.toasterService.showSnackBar("success", response?.body?.file);
+                            this.toasterService.showSnackBar("success", response?.body?.encodedData);
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/daybook/daybook.component.ts]: @nisha-walkover are you receiving message in encodedData key???

**Review by ravinderwalkover — CHANGES_REQUESTED** [review]

File: apps/web-giddh/src/app/daybook/daybook.component.ts (hunk 9 above)
Comment (nisha-walkover) [reply apps/web-giddh/src/app/daybook/daybook.component.ts]: yes sir

**Review by ravinderwalkover — APPROVED** [review]

**Pull request merged**